interceptor/tracing_interceptor.go

```diff
@@ -515,6 +546,77 @@ func (t *tracingWorkflowInboundInterceptor) HandleQuery(
 	return val, err
 }
 
+func (t *tracingWorkflowInboundInterceptor) ValidateUpdate(
+	ctx workflow.Context,
+	in *UpdateInput,
+) error {
+	// Only add tracing if enabled and not replaying
+	if t.root.options.DisableUpdateTracing || workflow.IsReplaying(ctx) {
+		return t.Next.ValidateUpdate(ctx, in)
+	}
+	// Start span reading from header
+	info := workflow.GetInfo(ctx)
+	currentUpdateInfo := workflow.GetCurrentUpdateInfo(ctx)
+	span, ctx, err := t.root.startSpanFromWorkflowContext(ctx, &TracerStartSpanOptions{
+		Operation: "ValidateUpdate",
+		Name:      in.Name,
+		Tags: map[string]string{
+			workflowIDTagKey: info.WorkflowExecution.ID,
+			runIDTagKey:      info.WorkflowExecution.RunID,
+			updateIDTagKey:   currentUpdateInfo.ID,
+		},
+		FromHeader: true,
+		Time:       time.Now(),
+		// We intentionally do not set IdempotencyKey here because validation is not run on
+		// replay. When the tracing interceptor's span counter is reset between workflow
+		// replays, the validator will not be processed which could result in impotency key
+		// collisions with other requests.
+	})
+	if err != nil {
+		return err
+	}
+	var finishOpts TracerFinishSpanOptions
+	defer span.Finish(&finishOpts)
+
+	err = t.Next.ValidateUpdate(ctx, in)
+	finishOpts.Error = err
+	return err
+}
+
+func (t *tracingWorkflowInboundInterceptor) ExecuteUpdate(
+	ctx workflow.Context,
+	in *UpdateInput,
+) (interface{}, error) {
+	// Only add tracing if enabled and not replaying
+	if t.root.options.DisableUpdateTracing || workflow.IsReplaying(ctx) {
+		return t.Next.ExecuteUpdate(ctx, in)
+	}
+	// Start span reading from header
+	info := workflow.GetInfo(ctx)
+	currentUpdateInfo := workflow.GetCurrentUpdateInfo(ctx)
+	span, ctx, err := t.root.startSpanFromWorkflowContext(ctx, &TracerStartSpanOptions{
+		Operation: "ExecuteUpdate",
```

Comment:
Hrmm, I see we didn't call this method `HandleUpdate` like we have `HandleSignal` and `HandleQuery` (and as we may have in other SDKs), but I wonder if we still want the span name to match the other SDKs or match the interceptor method.

Reply:
I'm fine matching other SDK's, I don't think the span name needs to match the interceptor name